internal/witness/witness.go

```diff
@@ -224,9 +224,9 @@ func (w witness) update(ctx context.Context, cp []byte, size uint64) ([]byte, er
 			if err != nil {
 				return nil, fmt.Errorf("witness at %q replied with x.tlog.size but body %q could not be parsed as decimal", w.url, bodyStr)
 			}
-			if newWitSize < int64(w.size) {
+			if newWitSize > int64(w.size) {
```

Comment:
```suggestion
			if newWitSize > int64(size) {
```

If you duplicated this block (i.e. leave the original if block unchanged) and then made this change to compare agains the size we just sent out (which should be the largest possible size for this log) then it would at least prevent one category of errors where a malicious actor could MitM the connection and respond with MaxUint64.